resources/qml/WelcomePages/AddPrinterByIpContent.qml

```diff
@@ -0,0 +1,229 @@
+
+    Item
+    {
+        anchors.top: titleLabel.bottom
+        anchors.bottom: connectButton.top
+        anchors.topMargin: UM.Theme.getSize("default_margin").height
+        anchors.bottomMargin: UM.Theme.getSize("default_margin").height
+        anchors.horizontalCenter: parent.horizontalCenter
+        width: Math.floor(parent.width * 3 / 4)
+
+        Item
+        {
+            width: parent.width
+
+            Label
+            {
+                id: explainLabel
+                height: contentHeight
+                width: parent.width
+                anchors.top: parent.top
+                anchors.margins: UM.Theme.getSize("default_margin").width
+                font: UM.Theme.getFont("default")
+
+                text: catalog.i18nc("@label", "Enter the IP address or hostname of your printer on the network.")
+            }
+
+            Item
+            {
+                id: userInputFields
+                height: childrenRect.height
+                width: parent.width
+                anchors.top: explainLabel.bottom
+
+                TextField
+                {
+                    id: hostnameField
+                    anchors.verticalCenter: addPrinterButton.verticalCenter
+                    anchors.left: parent.left
+                    height: addPrinterButton.height
+                    anchors.right: addPrinterButton.left
+                    anchors.margins: UM.Theme.getSize("default_margin").width
+                    font: UM.Theme.getFont("default")
+
+                    validator: RegExpValidator
+                    {
+                        regExp: /[a-fA-F0-9\.\:]*/
+                    }
+
+                    onAccepted: addPrinterButton.clicked()
+                }
+
+                Cura.PrimaryButton
+                {
+                    id: addPrinterButton
+                    anchors.top: parent.top
+                    anchors.right: parent.right
+                    anchors.margins: UM.Theme.getSize("default_margin").width
+                    width: UM.Theme.getSize("action_button").width
+                    fixedWidthMode: true
+
+                    text: catalog.i18nc("@button", "Add")
+                    onClicked:
+                    {
+                        if (hostnameField.text.trim() != "")
+                        {
+                            enabled = false;
+                            UM.OutputDeviceManager.addManualDevice(hostnameField.text, hostnameField.text);
+                        }
+                    }
+
+                    BusyIndicator
+                    {
+                        anchors.fill: parent
+                        running:
+                        {
+                            ! parent.enabled &&
+                            ! addPrinterByIpScreen.hasSentRequest &&
+                            ! addPrinterByIpScreen.haveConnection
+                        }
+                    }
+                }
+            }
+
+            Rectangle
```

Comment:
This shouldn't be a rectangle but an item (right now it doesn't set a background or do anything with rendering). if it should, explicitly set the color to something that is themable

Reply:
Fixed in 0c8afbfa4